packages/react-instantsearch-core-next/src/IndexContext.ts

```diff
@@ -0,0 +1,5 @@
+import { createContext } from 'react';
+
+import type { IndexWidget } from 'instantsearch.js/es/widgets/index/index';
+
+export const IndexContext = createContext<null | IndexWidget>(null);
```

Comment:
could we want to store anything else than the widget on the context? maybe in an object in that case?

Although YAGNI, it could be separate contexts so this is good!

Reply:
Yeah I think it would be separate contexts. Let's see how it evolves during the alpha!